Do not increase expected indent part way through a multi-line conditi…

p-behr opened the pull request:
Do not require continued lines to be indented.
The problem is that we are incrementing the expected indent as soon as we find a conditional (i.e. IF, WHEN) even if it's a continued line.  

### Changes
Defer incrementing the expected indent until we reach the end of the conditional statement.

### Checklist

* [x] have tested my change
* [ ] updated relevant documentation
* [x] Remove any/all `console.log`s I added
* [x] eslint is not complaining
* [ ] have added myself to the contributors' list in the README
* [x] **for feature PRs**: PR only includes one feature enhancement.

worksofliam commented:
@p-behr this is awesome. I need one more thing though: a unit test.

Please create one in the `tests/suite/basic.js`file. Take a look at other examples, I'm sure you will get the jist.

Then use `npm run test` to run them all.

p-behr commented:
> @p-behr this is awesome. I need one more thing though: a unit test.
> 
> Please create one in the `tests/suite/basic.js`file. Take a look at other examples, I'm sure you will get the gist.
> 
> Then use `npm run test` to run them all.

Totally didn't notice the tests in there; that's good to know and really cool!
I've added a couple tests for this change in linter.js (seemed like that's where they belong).  LinterWorker showed up when I ran all the tests so I changed that guy too.
Let me know if I'm missing anything!

worksofliam commented:
@p-behr This is awesome. Thank you!
